treebeard/mp_tree.py

```diff
@@ -373,8 +373,13 @@ def process(self):
                       ' increasing the path.max_length property'
                       ' and UPDATE your database'))
         else:
-            # adding the new child as the last one
-            newobj.path = self.node.get_last_child()._inc_path()
+            # adding the new child as the last one after checking if there is a Last Child
+            last_child = self.node.get_last_child():
+            newobj.path = (
+                last_child._inc_path()
+                if last_child is not None
+                else self.node.path + '0001'
```

Comment:
I'm not sure where you got this path format from - it should be via `self.node_cls._get_path(self.node.path, newobj.depth, 1)` right?

Reply:
it looks like what's happening here is the `self.numchild` attribute has been updated but the database doesn't have the child there